src/util/obfuscation.h

```diff
@@ -33,9 +34,26 @@ class Obfuscation
     {
         if (!*this) return;
 
-        const KeyType rot_key{m_rotations[key_offset % KEY_SIZE]}; // Continue obfuscation from where we left off
-        for (; target.size() >= KEY_SIZE; target = target.subspan(KEY_SIZE)) {
-            XorWord(target.first<KEY_SIZE>(), rot_key);
+        KeyType rot_key{m_rotations[key_offset % KEY_SIZE]}; // Continue obfuscation from where we left off
+        if (target.size() > KEY_SIZE) {
+            // Obfuscate until 64-bit alignment boundary
+            if (const auto misalign{std::bit_cast<uintptr_t>(target.data()) % KEY_SIZE}) {
```

Comment:
In commit "optimization: peel align-head and unroll body to 64 bytes" (248b6a27c351690d3596711cc36b8102977adeab)

reinterpret_cast seems more appropriate than bit_cast here, though either are probably ok. IIUC reinterpret cast is guaranteed to convert the pointer to an integer that can be round tripped back into a valid pointer, while bit_cast is looser and doesn't provide any guarantees.

Reply:
Can you please point me to a documentation or code which demonstrates that?

Reply:
Done in  [`fee3048` (#33039)](https://github.com/bitcoin/bitcoin/pull/33039/commits/fee3048fcc4972ba02127a2a4ef75524c5be275d)